// popup-modal/bundles/popup-modal.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('popup-modal', ['exports', '@angular/core'], factory) :
    (factory((global['popup-modal'] = {}),global.ng.core));
}(this, (function (exports,i0) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    var PopupModalService = (function () {
        function PopupModalService() {
            this.modals = [];
        }
        /**
         * @param {?} modal
         * @return {?}
         */
        PopupModalService.prototype.add = /**
         * @param {?} modal
         * @return {?}
         */
            function (modal) {
                this.modals.push(modal);
            };
        /**
         * @param {?} id
         * @return {?}
         */
        PopupModalService.prototype.remove = /**
         * @param {?} id
         * @return {?}
         */
            function (id) {
                this.modals = this.modals.filter(function (x) { return x.id !== id; });
            };
        /**
         * @param {?} id
         * @return {?}
         */
        PopupModalService.prototype.open = /**
         * @param {?} id
         * @return {?}
         */
            function (id) {
                /** @type {?} */
                var modal = this.modals.filter(function (x) { return x.id === id; })[0];
                modal.open();
            };
        /**
         * @param {?} id
         * @return {?}
         */
        PopupModalService.prototype.close = /**
         * @param {?} id
         * @return {?}
         */
            function (id) {
                /** @type {?} */
                var modal = this.modals.filter(function (x) { return x.id === id; })[0];
                modal.close();
            };
        PopupModalService.decorators = [
            { type: i0.Injectable, args: [{
                        providedIn: 'root'
                    },] },
        ];
        /** @nocollapse */
        PopupModalService.ctorParameters = function () { return []; };
        /** @nocollapse */ PopupModalService.ngInjectableDef = i0.defineInjectable({ factory: function PopupModalService_Factory() { return new PopupModalService(); }, token: PopupModalService, providedIn: "root" });
        return PopupModalService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    var PopupModalComponent = (function () {
        function PopupModalComponent(modalService, el) {
            this.modalService = modalService;
            this.el = el;
            this.element = el.nativeElement;
        }
        /**
         * @return {?}
         */
        PopupModalComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
                /** @type {?} */
                var modal = this;
                if (!this.id) {
                    console.error('Id is not available');
                    return;
                }
                document.body.appendChild(this.element);
                // close modal on background click
                this.element.addEventListener('click', function (e) {
                    if (e.target.className === 'ppl-modal') {
                        modal.close();
                    }
                });
                this.modalService.add(this);
            };
        /**
         * @return {?}
         */
        PopupModalComponent.prototype.ngOnDestroy = /**
         * @return {?}
         */
            function () {
                this.modalService.remove(this.id);
                this.element.remove();
            };
        // open modal
        /**
         * @return {?}
         */
        PopupModalComponent.prototype.open = /**
         * @return {?}
         */
            function () {
                this.element.style.display = 'block';
                document.body.classList.add('ppl-modal-open');
            };
        // close modal
        /**
         * @return {?}
         */
        PopupModalComponent.prototype.close = /**
         * @return {?}
         */
            function () {
                this.element.style.display = 'none';
                document.body.classList.remove('ppl-modal-open');
            };
        PopupModalComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'ppl-popup-modal',
                        template: "<div class=\"ppl-modal\">\n            <div class=\"ppl-modal-body\">\n                <ng-content></ng-content>\n            </div>\n        </div>\n        <div class=\"ppl-modal-background\"></div>",
                        styles: ["ppl-popup-modal{display:none}.ppl-modal{position:fixed;top:0;right:0;bottom:0;left:0;z-index:50;overflow:auto}.ppl-modal-background{position:fixed;top:0;right:0;bottom:0;left:0;background-color:#000;opacity:.8;z-index:10}.ppl-modal-body{padding:20px;background:#fff;margin:10%}body.ppl-modal-open{overflow:hidden}"],
                        encapsulation: i0.ViewEncapsulation.None
                    },] },
        ];
        /** @nocollapse */
        PopupModalComponent.ctorParameters = function () {
            return [
                { type: PopupModalService },
                { type: i0.ElementRef }
            ];
        };
        PopupModalComponent.propDecorators = {
            id: [{ type: i0.Input }]
        };
        return PopupModalComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    var PopupModalModule = (function () {
        function PopupModalModule() {
        }
        PopupModalModule.decorators = [
            { type: i0.NgModule, args: [{
                        imports: [],
                        declarations: [PopupModalComponent],
                        exports: [PopupModalComponent]
                    },] },
        ];
        return PopupModalModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */

    exports.PopupModalService = PopupModalService;
    exports.PopupModalComponent = PopupModalComponent;
    exports.PopupModalModule = PopupModalModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
