// popup-modal/esm2015/lib/popup-modal.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import { Component, ElementRef, Input, ViewEncapsulation } from '@angular/core';
import { PopupModalService } from './popup-modal.service';
export class PopupModalComponent {
    /**
     * @param {?} modalService
     * @param {?} el
     */
    constructor(modalService, el) {
        this.modalService = modalService;
        this.el = el;
        this.element = el.nativeElement;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        /** @type {?} */
        let modal = this;
        if (!this.id) {
            console.error('Id is not available');
            return;
        }
        document.body.appendChild(this.element);
        // close modal on background click
        this.element.addEventListener('click', function (e) {
            if (e.target.className === 'ppl-modal') {
                modal.close();
            }
        });
        this.modalService.add(this);
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        this.modalService.remove(this.id);
        this.element.remove();
    }
    /**
     * @return {?}
     */
    open() {
        this.element.style.display = 'block';
        document.body.classList.add('ppl-modal-open');
    }
    /**
     * @return {?}
     */
    close() {
        this.element.style.display = 'none';
        document.body.classList.remove('ppl-modal-open');
    }
}
PopupModalComponent.decorators = [
    { type: Component, args: [{
                selector: 'ppl-popup-modal',
                template: `<div class="ppl-modal">
            <div class="ppl-modal-body">
                <ng-content></ng-content>
            </div>
        </div>
        <div class="ppl-modal-background"></div>`,
                styles: [`ppl-popup-modal{display:none}.ppl-modal{position:fixed;top:0;right:0;bottom:0;left:0;z-index:50;overflow:auto}.ppl-modal-background{position:fixed;top:0;right:0;bottom:0;left:0;background-color:#000;opacity:.8;z-index:10}.ppl-modal-body{padding:20px;background:#fff;margin:10%}body.ppl-modal-open{overflow:hidden}`],
                encapsulation: ViewEncapsulation.None
            },] },
];
/** @nocollapse */
PopupModalComponent.ctorParameters = () => [
    { type: PopupModalService },
    { type: ElementRef }
];
PopupModalComponent.propDecorators = {
    id: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    PopupModalComponent.prototype.id;
    /** @type {?} */
    PopupModalComponent.prototype.element;
    /** @type {?} */
    PopupModalComponent.prototype.modalService;
    /** @type {?} */
    PopupModalComponent.prototype.el;
}
