// popup-modal/fesm2015/popup-modal.js
import { Injectable, Component, ElementRef, Input, ViewEncapsulation, NgModule, defineInjectable } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
class PopupModalService {
    constructor() {
        this.modals = [];
    }
    /**
     * @param {?} modal
     * @return {?}
     */
    add(modal) {
        this.modals.push(modal);
    }
    /**
     * @param {?} id
     * @return {?}
     */
    remove(id) {
        this.modals = this.modals.filter(x => x.id !== id);
    }
    /**
     * @param {?} id
     * @return {?}
     */
    open(id) {
        /** @type {?} */
        let modal = this.modals.filter(x => x.id === id)[0];
        modal.open();
    }
    /**
     * @param {?} id
     * @return {?}
     */
    close(id) {
        /** @type {?} */
        let modal = this.modals.filter(x => x.id === id)[0];
        modal.close();
    }
}
PopupModalService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] },
];
/** @nocollapse */
PopupModalService.ctorParameters = () => [];
/** @nocollapse */ PopupModalService.ngInjectableDef = defineInjectable({ factory: function PopupModalService_Factory() { return new PopupModalService(); }, token: PopupModalService, providedIn: "root" });

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
class PopupModalComponent {
    /**
     * @param {?} modalService
     * @param {?} el
     */
    constructor(modalService, el) {
        this.modalService = modalService;
        this.el = el;
        this.element = el.nativeElement;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        /** @type {?} */
        let modal = this;
        if (!this.id) {
            console.error('Id is not available');
            return;
        }
        document.body.appendChild(this.element);
        // close modal on background click
        this.element.addEventListener('click', function (e) {
            if (e.target.className === 'ppl-modal') {
                modal.close();
            }
        });
        this.modalService.add(this);
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        this.modalService.remove(this.id);
        this.element.remove();
    }
    /**
     * @return {?}
     */
    open() {
        this.element.style.display = 'block';
        document.body.classList.add('ppl-modal-open');
    }
    /**
     * @return {?}
     */
    close() {
        this.element.style.display = 'none';
        document.body.classList.remove('ppl-modal-open');
    }
}
PopupModalComponent.decorators = [
    { type: Component, args: [{
                selector: 'ppl-popup-modal',
                template: `<div class="ppl-modal">
            <div class="ppl-modal-body">
                <ng-content></ng-content>
            </div>
        </div>
        <div class="ppl-modal-background"></div>`,
                styles: [`ppl-popup-modal{display:none}.ppl-modal{position:fixed;top:0;right:0;bottom:0;left:0;z-index:50;overflow:auto}.ppl-modal-background{position:fixed;top:0;right:0;bottom:0;left:0;background-color:#000;opacity:.8;z-index:10}.ppl-modal-body{padding:20px;background:#fff;margin:10%}body.ppl-modal-open{overflow:hidden}`],
                encapsulation: ViewEncapsulation.None
            },] },
];
/** @nocollapse */
PopupModalComponent.ctorParameters = () => [
    { type: PopupModalService },
    { type: ElementRef }
];
PopupModalComponent.propDecorators = {
    id: [{ type: Input }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
class PopupModalModule {
}
PopupModalModule.decorators = [
    { type: NgModule, args: [{
                imports: [],
                declarations: [PopupModalComponent],
                exports: [PopupModalComponent]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */

export { PopupModalService, PopupModalComponent, PopupModalModule };
