// popup-modal/esm5/lib/popup-modal.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
var PopupModalService = /** @class */ (function () {
    function PopupModalService() {
        this.modals = [];
    }
    /**
     * @param {?} modal
     * @return {?}
     */
    PopupModalService.prototype.add = /**
     * @param {?} modal
     * @return {?}
     */
    function (modal) {
        this.modals.push(modal);
    };
    /**
     * @param {?} id
     * @return {?}
     */
    PopupModalService.prototype.remove = /**
     * @param {?} id
     * @return {?}
     */
    function (id) {
        this.modals = this.modals.filter(function (x) { return x.id !== id; });
    };
    /**
     * @param {?} id
     * @return {?}
     */
    PopupModalService.prototype.open = /**
     * @param {?} id
     * @return {?}
     */
    function (id) {
        /** @type {?} */
        var modal = this.modals.filter(function (x) { return x.id === id; })[0];
        modal.open();
    };
    /**
     * @param {?} id
     * @return {?}
     */
    PopupModalService.prototype.close = /**
     * @param {?} id
     * @return {?}
     */
    function (id) {
        /** @type {?} */
        var modal = this.modals.filter(function (x) { return x.id === id; })[0];
        modal.close();
    };
    PopupModalService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] },
    ];
    /** @nocollapse */
    PopupModalService.ctorParameters = function () { return []; };
    /** @nocollapse */ PopupModalService.ngInjectableDef = i0.defineInjectable({ factory: function PopupModalService_Factory() { return new PopupModalService(); }, token: PopupModalService, providedIn: "root" });
    return PopupModalService;
}());
export { PopupModalService };
if (false) {
    /** @type {?} */
    PopupModalService.prototype.modals;
}
