// popup-modal/esm5/lib/popup-modal.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import { Component, ElementRef, Input, ViewEncapsulation } from '@angular/core';
import { PopupModalService } from './popup-modal.service';
var PopupModalComponent = /** @class */ (function () {
    function PopupModalComponent(modalService, el) {
        this.modalService = modalService;
        this.el = el;
        this.element = el.nativeElement;
    }
    /**
     * @return {?}
     */
    PopupModalComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var modal = this;
        if (!this.id) {
            console.error('Id is not available');
            return;
        }
        document.body.appendChild(this.element);
        // close modal on background click
        this.element.addEventListener('click', function (e) {
            if (e.target.className === 'ppl-modal') {
                modal.close();
            }
        });
        this.modalService.add(this);
    };
    /**
     * @return {?}
     */
    PopupModalComponent.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        this.modalService.remove(this.id);
        this.element.remove();
    };
    // open modal
    /**
     * @return {?}
     */
    PopupModalComponent.prototype.open = /**
     * @return {?}
     */
    function () {
        this.element.style.display = 'block';
        document.body.classList.add('ppl-modal-open');
    };
    // close modal
    /**
     * @return {?}
     */
    PopupModalComponent.prototype.close = /**
     * @return {?}
     */
    function () {
        this.element.style.display = 'none';
        document.body.classList.remove('ppl-modal-open');
    };
    PopupModalComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ppl-popup-modal',
                    template: "<div class=\"ppl-modal\">\n            <div class=\"ppl-modal-body\">\n                <ng-content></ng-content>\n            </div>\n        </div>\n        <div class=\"ppl-modal-background\"></div>",
                    styles: ["ppl-popup-modal{display:none}.ppl-modal{position:fixed;top:0;right:0;bottom:0;left:0;z-index:50;overflow:auto}.ppl-modal-background{position:fixed;top:0;right:0;bottom:0;left:0;background-color:#000;opacity:.8;z-index:10}.ppl-modal-body{padding:20px;background:#fff;margin:10%}body.ppl-modal-open{overflow:hidden}"],
                    encapsulation: ViewEncapsulation.None
                },] },
    ];
    /** @nocollapse */
    PopupModalComponent.ctorParameters = function () { return [
        { type: PopupModalService },
        { type: ElementRef }
    ]; };
    PopupModalComponent.propDecorators = {
        id: [{ type: Input }]
    };
    return PopupModalComponent;
}());
export { PopupModalComponent };
if (false) {
    /** @type {?} */
    PopupModalComponent.prototype.id;
    /** @type {?} */
    PopupModalComponent.prototype.element;
    /** @type {?} */
    PopupModalComponent.prototype.modalService;
    /** @type {?} */
    PopupModalComponent.prototype.el;
}
